app/assets/javascripts/basetypes/Code.js

```diff
@@ -9,19 +9,19 @@ Code.prototype = Object.create(mongoose.SchemaType.prototype);
 Code.prototype.cast = (code) => {
   if (code != null) {
     // handles codes that have not yet been cast to a code and those that have already been cast to a code
-    if (code.code && (code.codeSystemOid || code.system)) {
+    if (code.code && code.system) {
```

Comment:
The first check should be simply `if (code.isCode) { return code; }`. This would make it so no casting would be done for things that are already codes.